bundles/org.openhab.binding.smsmodem/src/main/java/org/openhab/binding/smsmodem/internal/handler/SMSModemBridgeHandler.java

```diff
@@ -0,0 +1,473 @@
+        this.serialPortManager = serialPortManager;
+    }
+
+    @Override
+    protected void updateConfiguration(Configuration configuration) {
+        super.updateConfiguration(configuration);
+        scheduler.execute(() -> {
+            modem.stop();
+            checkAndStartModemIfNeeded();
+        });
+    }
+
+    @Override
+    public void initialize() {
+        shouldRun = true;
+        if (checkScheduled == null || (checkScheduled.isDone()) && this.shouldRun) {
+            checkScheduled = scheduler.scheduleWithFixedDelay(this::checkAndStartModemIfNeeded, 0, 15,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    private void checkAndStartModemIfNeeded() {
+        try {
+            if (shouldRun && !isRunning()) {
+                logger.debug("Initializing smsmodem");
+                // ensure the underlying modem is stopped before trying to (re)starting it :
+                if (modem != null) {
+                    modem.stop();
+                }
+                String logName;
+                if (getThing().getThingTypeUID().equals(SMSModemBindingConstants.SMSMODEMBRIDGE_THING_TYPE)) {
+                    SMSModemBridgeConfiguration config = getConfigAs(SMSModemBridgeConfiguration.class);
+                    modem = new Modem(serialPortManager, resolveEventualSymbolicLink(config.serialPort),
+                            Integer.valueOf(config.baud), config.simPin, scheduler, config.pollingInterval,
+                            config.delayBetweenSend);
+                    checkParam(config);
+                    logName = config.serialPort + " | " + config.baud;
+                } else if (getThing().getThingTypeUID()
+                        .equals(SMSModemBindingConstants.SMSMODEMREMOTEBRIDGE_THING_TYPE)) {
+                    SMSModemRemoteBridgeConfiguration config = getConfigAs(SMSModemRemoteBridgeConfiguration.class);
+                    modem = new Modem(serialPortManager, resolveEventualSymbolicLink(config.ip),
+                            Integer.valueOf(config.networkPort), config.simPin, scheduler, config.pollingInterval,
+                            config.delayBetweenSend);
+                    checkRemoteParam(config);
+                    logName = config.ip + ":" + config.networkPort;
+                } else {
+                    throw new IllegalArgumentException("Invalid thing type");
+                }
+                logger.debug("Now trying to start SMSModem {}", logName);
+                modem.registerStatusListener(this);
+                modem.registerMessageListener(this);
+                modem.registerInformationListener(this);
+                modem.start();
+                logger.debug("SMSModem {} started", logName);
+            }
+        } catch (ModemConfigurationException e) {
+            String message = e.getMessage();
+            logger.error(message, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, message);
```

Comment:
Is this rather a CONFIGURATION_ERROR?

Reply:
Both !
A misconfiguration (bad address) or network/serial error prevents from communicating with the modem, which is not joinable at the address set.
So, what should I use ?

Reply:
This is what I saw. That sounds more like a configuration error. I leave it up to you what makes more sense.
![grafik](https://user-images.githubusercontent.com/1894841/202898901-0b5f44ea-0a30-4517-bac4-364ae0c7b677.png)

Reply:
You are right.
I splitted it in two, depending on the root cause (IOException = communication error, other = configuration error).
Done